drama-queen/src/ui/components/orchestrator/tools/useAutoNext.ts

```diff
@@ -13,25 +13,34 @@ type PartialLunatic = Pick<
 export function useAutoNext() {
   const ref = useRef<PartialLunatic | null>(null)
 
-  const onChange = useCallback((valueChange: { name: string }) => {
-    if (ref.current === null) return
-    const { getComponents, goNextPage } = ref.current
-
-    const variableChanged = valueChange?.name
-    const components = getComponents()
-    const firstComponent = components[0]
-    if (
-      // There is only one "don't know / refusal" variable on the page
-      countMissingResponseInPage(components) === 1 &&
-      // The value changed is a "don't know / refusal" response, or the current component is a radio or checkbox
-      (variableChanged.includes('_MISSING') ||
-        ['Radio', 'CheckboxBoolean', 'CheckboxOne'].includes(
-          firstComponent.componentType
-        ))
-    ) {
-      goNextPage()
-    }
-  }, [])
+  const onChange = useCallback(
+    (
+      valueChange: {
+        name: string
+        value: any
+        iteration?: number[]
+      }[]
+    ) => {
+      if (ref.current === null) return
+      const { getComponents, goNextPage } = ref.current
+      const variablesChanged = valueChange[0].name
```

Comment:
it's not a good point to check only the first changed value, you should look at all and break at first goNext